spec.html

```diff
@@ -30871,160 +30871,24 @@
-        <emu-alg>
-          1. Assert: Type(_object_) is Object and _object_ does not have either a [[TypedArrayName]] or an [[ArrayBufferData]] internal slot.
-          1. If NewTarget is *undefined*, throw a *TypeError* exception.
-          1. Return TypedArrayFrom(NewTarget, _object_, *undefined*, *undefined*).
-        </emu-alg>
-      </emu-clause>
-
-      <!-- es6num="22.2.1.5" -->
-      <emu-clause id="sec-%typedarray%-buffer-byteoffset-length">
-        <h1>%TypedArray% ( _buffer_ [ , _byteOffset_ [ , _length_ ] ] )</h1>
-        <p>This description applies only if the %TypedArray% function is called with at least one argument and the Type of the first argument is Object and that object has an [[ArrayBufferData]] internal slot.</p>
-        <p>%TypedArray% called with arguments _buffer_, _byteOffset_, and _length_ performs the following steps:</p>
+      <!-- es6num="22.2.4.1" -->
+      <emu-clause id="sec-typedarray">
+        <h1>%TypedArray%()</h1>
+        <p>The %TypedArray% constructor performs the following steps:</p>
         <emu-alg>
-          1. Assert: Type(_buffer_) is Object and _buffer_ has an [[ArrayBufferData]] internal slot.
           1. If NewTarget is *undefined*, throw a *TypeError* exception.
-          1. Let _O_ be ? AllocateTypedArray(NewTarget).
-          1. Let _constructorName_ be the String value of _O_'s [[TypedArrayName]] internal slot.
-          1. Let _elementSize_ be the Number value of the Element Size value in <emu-xref href="#table-49"></emu-xref> for _constructorName_.
-          1. Let _offset_ be ? ToInteger(_byteOffset_).
-          1. If _offset_ &lt; 0, throw a *RangeError* exception.
-          1. If _offset_ is -0, let _offset_ be +0.
-          1. If _offset_ modulo _elementSize_ &ne; 0, throw a *RangeError* exception.
-          1. If IsDetachedBuffer(_buffer_) is *true*, throw a *TypeError* exception.
-          1. Let _bufferByteLength_ be the value of _buffer_'s [[ArrayBufferByteLength]] internal slot.
-          1. If _length_ is *undefined*, then
-            1. If _bufferByteLength_ modulo _elementSize_ &ne; 0, throw a *RangeError* exception.
-            1. Let _newByteLength_ be _bufferByteLength_ - _offset_.
-            1. If _newByteLength_ &lt; 0, throw a *RangeError* exception.
-          1. Else,
-            1. Let _newLength_ be ? ToLength(_length_).
-            1. Let _newByteLength_ be _newLength_ &times; _elementSize_.
-            1. If _offset_+_newByteLength_ &gt; _bufferByteLength_, throw a *RangeError* exception.
-          1. Set _O_'s [[ViewedArrayBuffer]] internal slot to _buffer_.
-          1. Set _O_'s [[ByteLength]] internal slot to _newByteLength_.
-          1. Set _O_'s [[ByteOffset]] internal slot to _offset_.
-          1. Set _O_'s [[ArrayLength]] internal slot to _newByteLength_ / _elementSize_ .
-          1. Return _O_.
+          1. Let _here_ be the active function.
+          1. Let _super_ be ? _here_.[[GetPrototypeOf]]().
```

Comment:
You need the line I mentioned in https://github.com/tc39/ecma262/issues/163#issuecomment-158116899 :

1. If SameValue(NewTarget, _here_) is **true**, throw a **TypeError** exception.

This is necessary to prevent things like:

```js
 let obj = new Int32Array.__proto__;
```